Мои изменения

n0-n4-m3 opened the pull request:
Все что в моих коммитах я добавил

n0-n4-m3 committed:
Проверка времени, проверка ФИО, устойчивость

n0-n4-m3 committed:
Вернул API токен основного бота

n0-n4-m3 committed:
update API token

убрал лишний символ из токена...

TIMAIGER64 reviewed (commented):
ВСЁ ОХУЕННО ОБЪЕДИНЯЙ

Georgii7725 commented on main.py:
```diff
@@ -1,3 +1,4 @@
+from turtle import home
```
Убрать

Georgii7725 commented on main.py:
```diff
@@ -18,15 +19,17 @@ async def get_command_start(message: types.Message):
 
 # HELP COMMAND
 @dp.callback_query_handler(text=['HELP'])
-@dp.message_handler(text=['HELP'])
```
Нужно оставить

Georgii7725 commented on main.py:
```diff
@@ -18,15 +19,17 @@ async def get_command_start(message: types.Message):
     <b>REGISTER</b> - <em>регистрация ученика (при условии проживания в общежитии АГ)</em>\n<b>LOGIN</b> - <em>вход в аккаунт ученика</em>"""
+
+    await bot.answer_callback_query(callback.id)
     
```
Согласен, надо оставить

Georgii7725 commented on main.py:
```diff
@@ -18,15 +19,17 @@ async def get_command_start(message: types.Message):
                             text=help_answer,
-                            parse_mode="HTML")
+                            parse_mode="HTML",
+                            reply_markup=kb)
```
Нужно оставить клаву

n0-n4-m3 replied:
Как мне сказал тимур то когда ты заходишь в help, то клава не меняется и можно использовать старую

Georgii7725 commented on main.py:
```diff
@@ -66,21 +69,31 @@ async def add_new_user(callback: types.CallbackQuery):
+                            text=text)
+        await register.num_room.set()
+    except:
+        text="""<b>Даун пиши ФИО правильно</b>"""
```
Нормальный текст нужен

Georgii7725 commented on main.py:
```diff
@@ -110,16 +123,15 @@ async def state3(message: types.Message, state=FSMContext):
     if data['check_code_reg'] == reg_active_code:
         text="""<b>Регистрация прошла успешно.</b> \n<em>Система зафиксировала, что ты являешься учеником АГ!</em>"""
         FIO = data['FIO']
-        FIO = FIO.split()
```
Согласен

Georgii7725 commented on main.py:
```diff
@@ -93,7 +106,7 @@ async def state2(message: types.Message, state=FSMContext):
+async def st(message: types.Message, state=FSMContext):
     
     ans = message.text  
     global reg_active_code
```
Все global убрать, добавить в state

Georgii7725 commented on main.py:
```diff
@@ -160,16 +172,17 @@ async def state_log(message: types.Message, state=FSMContext):
     if flag:
         global log_active_code
         log_active_code = send_mail(ans)
+        await state.update_data(st_log=ans)
```
Согласен

Georgii7725 commented on main.py:
```diff
@@ -233,66 +246,73 @@ async def get_command_login(message: types.Message):
                             text=text,
                             parse_mode="HTML",
                             reply_markup=ReplyKeyboardRemove())
-    await exit.exit_time.set()
```
Нужно оставить

Georgii7725 commented on imports.py:
```diff
@@ -3,19 +3,21 @@
 dp = Dispatcher(bot, storage = MemoryStorage())
 
-
+global hour_entrance_int
```
Убрать

n0-n4-m3 committed:
Всё замерждил

Добавил дату, все проверки, избавился от глобал, починил help, Убрал логин и логаут

n0-n4-m3 commented:
Короче, как ты сделал так нихера не работает в help. Для callback_querry свой handler для message_handler свой handler поэтому вот. Если тебе так не нравится, ну меняй как нравится там не так много
